core/src/main/java/org/geysermc/geyser/bedrock/camera/GeyserCameraFade.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2019-2023 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.bedrock.camera;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.geysermc.geyser.api.bedrock.camera.CameraFade;
+
+import java.awt.Color;
+
+public record GeyserCameraFade(
+        Color color,
+        float fadeInSeconds,
+        float holdSeconds,
+        float fadeOutSeconds
+) implements CameraFade {
+
+    public static class CustomFadeBuilder implements CameraFade.Builder {
```

Comment:
can this be `Builder` ?

Reply:
it can be, but it wouldn't be consistent with other builder impls, such as https://github.com/GeyserMC/Geyser/blob/master/core/src/main/java/org/geysermc/geyser/item/GeyserCustomItemOptions.java#L40

Reply:
i understand but i also think that those previous ones should be renamed :P

the subclass isn't statically imported anywhere and its weird to prefix nested classes with what is basically the class name (because nesting it already affects the namespace)

Reply:
yeah, fair point; ill change